source/common/formatter/substitution_formatter.cc

```diff
@@ -1700,14 +1714,29 @@ GrpcStatusFormatter::format(const Http::RequestHeaderMap&,
   if (!grpc_status.has_value()) {
     return absl::nullopt;
   }
-  if (format_as_number_) {
-    return std::to_string(grpc_status.value());
+  switch (format_) {
+  case CamelString: {
+    const auto grpc_status_message = Grpc::Utility::grpcStatusToString(grpc_status.value());
+    if (grpc_status_message == EMPTY_STRING || grpc_status_message == "InvalidCode") {
+      return std::to_string(grpc_status.value());
+    }
+    return grpc_status_message;
+  }
+  case SnakeString: {
+    const auto grpc_status_message =
+        absl::StatusCodeToString(static_cast<absl::StatusCode>(grpc_status.value()));
+    if (grpc_status_message == EMPTY_STRING) {
+      return std::to_string(grpc_status.value());
+    }
+    return grpc_status_message;
   }
-  const auto grpc_status_message = Grpc::Utility::grpcStatusToString(grpc_status.value());
-  if (grpc_status_message == EMPTY_STRING || grpc_status_message == "InvalidCode") {
+  case Number: {
     return std::to_string(grpc_status.value());
   }
-  return grpc_status_message;
+  default:
+    throw EnvoyException("Only CAMEL_STRING, SNAKE_STRING and NUMBER are "
+                         "supported in grpc status access log format.");
+  }
```

Comment:
May be using `PANIC_DUE_TO_CORRUPT_ENUM` here? Because this runtime exception will not be caught, this branch should never be reached. The `GrpcStatusFormatter::parseFormat` ensure that.

Reply:
Done.